runner/execer/os/os_execer.go

```diff
@@ -193,7 +194,9 @@ func (e *osExecer) monitorMem(p *osProcess) {
 						"jobID":       p.JobID,
 						"taskID":      p.TaskID,
 					}).Infof("Increased mem_cap utilization for pid %d to %d", pid, int(memUsagePct*100))
-				ps, err := exec.Command("ps", "-u", os.Getenv("USER"), "-opid,sess,ppid,pgid,rss,args").CombinedOutput()
+
+				ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+				ps, err := exec.CommandContext(ctx, "ps", "-u", os.Getenv("USER"), "-opid,sess,ppid,pgid,rss,args").CombinedOutput()
```

Comment:
Recommend logging ctx.Err() if it == context.DeadlineExceeded so we know this is happening